### PR TITLE
Canvas: Add metric value element type

### LINKED ISSUE #55121: Canvas: Add metric value element type
Add a new element to canvas "Metric value" that is a modified text box that defaults to displaying a selection for what field value to display 

This should help users find value from panel as quickly as possible - potentially make this element type the initial default element that shows up on a new canvas panel

### LINKED ISSUE #55328: Canvas: Double click metric value to reset value
When inline editing is enabled, capture double click event on a set metric value element to "reset" it back to the initial state of the field picker

### PR DESCRIPTION
**Pull request by nmarrs**
**What this PR does / why we need it**:
Add new element type to canvas - `Metric Value`

This element allows user to easily select a field value to display on the panel without ever touching the editor. Ideally this element type will simplify basic workflow of displaying data on the canvas panel.

Current UX - double click element to set / reset field

https://user-images.githubusercontent.com/22381771/191379304-c007b5d4-08c8-4a1b-a5d0-6b82a7f1e8d7.mov

**Which issue(s) this PR fixes**:

Fixes #55121
Fixes #55328

### CODE REVIEW

File: packages/grafana-ui/src/components/MatchersUI/FieldNamePicker.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Select } from '../Select/Select';
 
 import { useFieldDisplayNames, useSelectOptions, frameHasName } from './utils';
 
-// Pick a field name out of the fulds
+// Pick a field name out of the fields
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/MatchersUI/FieldNamePicker.tsx:9]: fulds FTW!!!

File: public/app/features/canvas/runtime/scene.tsx [hunk 2]
```diff
@@ -67,13 +67,23 @@ export class Scene {
...
   constructor(
     cfg: CanvasFrameOptions,
     enableEditing: boolean,
     showAdvancedTypes: boolean,
     public onSave: (cfg: CanvasFrameOptions) => void
   ) {
     this.root = this.load(cfg, enableEditing, showAdvancedTypes);
+
+    // TODO: Unsubscribe from this (cleanup?)
+    this.editModeEnabled.subscribe((open) => {
```
Comment (ryantxu) [inline public/app/features/canvas/runtime/scene.tsx:81]:
```suggestion
    this.subscription = this.editModeEnabled.subscribe((open) => {
```
and then later in some shutdown...
````
this.subscription.unsubscribe()
```

:)

File: public/app/features/canvas/runtime/element.tsx [hunk 3]
```diff
@@ -445,10 +445,19 @@ export class ElementState implements LayerElement {
   };
 
   render() {
-    const { item } = this;
+    const { item, div } = this;
+    const scene = this.getScene();
+    // TODO: Rethink selected state handling
```
Comment (ryantxu) [inline public/app/features/canvas/runtime/element.tsx:450]: 😬 this is the piece I am a bit nervous about fixing later 

does the `div` have any attributes that can be inspected? (also not great, but better than the tree walking and scanning with every render)

Comment (nmarrs) [reply public/app/features/canvas/runtime/element.tsx:450]: With selecto we could in theory add a class to the selected element(s) divs and remove it on un-selection (see [example here](https://daybrush.com/selecto/release/latest/doc/Selecto.html#.event:select)). Not sure this helps us too much though given that adds an additional layer of complexity vs. just getting current selection state from selecto itself.

> does the `div` have any attributes that can be inspected? (also not great, but better than the tree walking and scanning with every render)

This current implementation doesn't tree walk / scan with every render as it just retrieves a set property on the selecto object (so in theory pretty minimal performance hit) (see implementation details of `getSelectedTargets` [here](https://daybrush.com/selecto/release/latest/doc/packages_selecto_src_SelectoManager.tsx.html#line229))

Is our concern that the element should know / have control over its own selected state vs. just managing it through selecto? I think the current approach is viable, at least for now, though I could be missing something. Open to discussing this more though

Comment (ryantxu) [reply public/app/features/canvas/runtime/element.tsx:450]: it is fine for now... but selection handling feels like the most fragile/finicky bits 😬

**Review by ryantxu — APPROVED** [review]

**Pull request merged**